R/snapshot.R

```diff
@@ -458,3 +465,20 @@ check_variant <- function(x) {
     abort("If supplied, `variant` must be a string")
   }
 }
+
+with_is_snapshotting <- function(code) {
+  withr::local_options(testthat_is_snapshotting = TRUE)
+  code
```

Comment:
I don't necessarily think this matters but looking at `with_` functions I have implemented, I seem to do it like this:

``` r
with_is_snapshotting <- function(code) {
  withr::with_options(list(testthat_is_snapshotting = TRUE), code = code)
}
```

I guess maybe the one difference is whether you `force(code)` or not?

Reply:
I prefer the `local_` syntax pretty much everywhere.